prov/shm/src/smr_atomic.c

```diff
@@ -414,6 +414,7 @@ static ssize_t smr_atomic_inject(struct fid_ep *ep_fid, const void *buf,
 	ofi_cirque_commit(smr_cmd_queue(peer_smr));
 	peer_smr->cmd_cnt--;
 
+	ret = smr_complete_tx(ep, NULL, ofi_op_atomic, 0, 0);
```

Comment:
I think the fix here is to call smr_cntr_report_tx_comp() rather than smr_compete_tx().  That makes it clear that only a counter is being updated, and we avoid the extra checks in smr_complete_tx() that we expect to be true.